builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +62,51 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
-	// Start a transaction
-	tx, err := db.Begin()
+	// Get the role
+	pathParts := strings.Split(req.Path, "/")
+	if len(pathParts) < 1 {
+		return nil, fmt.Errorf("Role name could not be determined")
+	}
+	name := pathParts[len(pathParts)-1]
+	role, err := b.Role(req.Storage, name)
 	if err != nil {
 		return nil, err
 	}
-	defer tx.Rollback()
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", name)), nil
+	}
 
-	// Revoke all permissions for the user. This is done before the
-	// drop, because MySQL explicitly documents that open user connections
-	// will not be closed. By revoking all grants, at least we ensure
-	// that the open connection is useless.
-	_, err = tx.Exec("REVOKE ALL PRIVILEGES, GRANT OPTION FROM '" + username + "'@'%'")
+	// Start a transaction
+	tx, err := db.Begin()
 	if err != nil {
 		return nil, err
 	}
+	defer tx.Rollback()
+
+	for _, query := range strutil.ParseArbitraryStringSlice(role.RevokeSQL, ";") {
```

Comment:
You have no default case here; if no role revocation SQL exists it should use the same logic as before, including dropping the user. If you want to turn that into a format that works with this logic (e.g. `if role.RevokeSQL == ""` set `role.RevokeSQL` to those two statements in a form that the `for` loop will execute correctly), that seems like a good option.